### PR TITLE
MM-10717: Update longpress buttons when add reaction, edit post, or d…

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary
Updates post longpress actions when a permissions for add reaction, edit post, or delete post are changed.

#### Ticket Link
[MM-10717](https://mattermost.atlassian.net/browse/MM-10717)

#### Checklist
n/a

#### Device Information
iPhon 6, iOS 11.4 (emulated)

#### Screenshots
https://youtu.be/FDdsZ4rBrVw

### CODE REVIEW

File: app/components/post_body/index.js [hunk 1]
```diff
@@ -52,6 +57,20 @@ function mapStateToProps(state, ownProps) {
     const isUserCanManageMembers = canManageChannelMembers(state);
     const isEphemeralPost = isPostEphemeral(post);
 
+    const {config, license} = state.entities.general;
```
Comment (saturninoabril) [inline app/components/post_body/index.js]: We have selectors for these, like 
``import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';``

File: app/components/tooltip.js [hunk 2]
```diff
@@ -39,6 +40,13 @@ export default class ToolTip extends PureComponent {
         }
     };
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.actions.length !== this.props.actions.length && getToolTipVisible()) {
```
Comment (saturninoabril) [inline app/components/tooltip.js]: If `actions` is optional then calling length of undefined will throw an error.

Comment (mkraft) [reply app/components/tooltip.js]: I think it should be required.

File: app/components/post_body/index.js (hunk 1 above)
Comment (mkraft) [reply app/components/post_body/index.js]: I have switched all instances of direct access of `license` and `config` from state to use selectors.

**Comment by mkraft** [comment]
@enahum It doesn't work in Android and the [react-native-bottom-sheet](https://github.com/WhatAKitty/react-native-bottom-sheet) package does not expose methods to programatically close or open the native Android `BottomSheet` package. It looks to me like we would need to fork the package and add methods something like:
```java
@ReactMethod
public void expand() {
    ...
    BottomSheetBehavior bottomSheetBehavior = BottomSheetBehavior.from(mLayoutBottomSheet);
    mbottomSheetBehavior.setState(BottomSheetBehavior.STATE_EXPANDED);
}

@ReactMethod
public void collapse() {
    ...
    BottomSheetBehavior bottomSheetBehavior = BottomSheetBehavior.from(mLayoutBottomSheet);
    mbottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);
}

@ReactMethod
public boolean isExpanded() {
    ...
    BottomSheetBehavior bottomSheetBehavior = BottomSheetBehavior.from(mLayoutBottomSheet);
    return buttomSheetBehaviour.getState() == BottomSheetBehavior.STATE_EXPANDED;
}
```

**Comment by mkraft** [comment]
@enahum Since the original issue was that the app needed to be killed and reopened to show changes to those buttons we have solved that for iOS and Android. The iOS "updating while still open" was a bonus. With that in mind I think we should add a new story to get that bonus feature working in Android. Sound good?

**Comment by enahum** [comment]
@mkraft yeah I'm ok with that.. please create the ticket :)

going to merge this one

**Pull request merged**